### ISSUE TITLE
False positive for subscripts in operator_whitespace

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#594]
Hi, I just found that the Swift Linter marks the following line as a warning with the `operator_whitespace` rule turned on:

```
// warning: Operator Function Whitespace Violation: Operators should be surrounded by a single whitespace when defining them. (operator_whitespace)
public func ==(lhs: Answer, rhs: Answer) -> Bool {
    return lhs.string == rhs.string
}
```

If I add a whitespace after the `==` then it goes away:

``` Swift
// no warnings or errors
public func == (lhs: Answer, rhs: Answer) -> Bool {
    return lhs.string == rhs.string
}
```

I think this is a false positive as subscripts are functions and therefore the function syntax should apply (without spaces after the function name), don't you think? Although I can turn the rule off for that line I think it should also be resolved within the rule.

**Comment by norio-nomura** [comment]
@jpsim: Is this a bug?

As reading [OperatorFunctionWhitespaceRule.swift](https://github.com/realm/SwiftLint/blob/master/Source%2FSwiftLintFramework%2FRules%2FOperatorFunctionWhitespaceRule.swift#L18-L20), it seems that behavior is just the Function of `operator_whitespace` rule.
> identifier: "operator_whitespace",
 name: "Operator Function Whitespace",
 description: "Operators should be surrounded by a single whitespace when defining them.",

**Comment by jpsim** [comment]
You're right @norio-nomura, this is intended behavior following the GitHub Swift guidelines that SwiftLint is built for.
